server/logger.js

```diff
@@ -0,0 +1,100 @@
+import winston from 'winston';
+import fs from 'fs';
+import { fileURLToPath } from 'url';
+import path from 'path';
+
+const { combine, timestamp, printf, colorize } = winston.format;
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const ROOT = __dirname;
+
+// Apply colors to log levels
+const colorizer = colorize({ all: true });
+
+const myFormat_console = printf(({ level, message, timestamp }) => {
+  return `${timestamp} ${level}: ${message}`;
+});
+
+const logger = new winston.createLogger({
+  level: 'info',
+  defaultMeta: { service: 'user-service' },
+  transports: [
+    new winston.transports.Console({
+      format: combine(colorizer, timestamp(), myFormat_console),
+    }),
+    new winston.transports.File({
+      filename: 'server.log',
+      format: combine(timestamp(), myFormat_console),
+    }),
+  ],
+  exitOnError: false,
+});
+
+fs.writeFile('server.log', '', () => {});
+
+function formatLogArguments(args) {
+  args = Array.prototype.slice.call(args);
+  var stackInfo = getStackInfo(1);
+  if (stackInfo) {
+    // get file path relative to project root
+    var calleeStr =
+      '[' +
+      stackInfo.relativePath +
+      ':' +
+      stackInfo.line +
+      ':' +
+      stackInfo.pos +
+      ']';
+    if (typeof args[0] === 'string') {
+      args[0] = calleeStr + ' ' + args[0];
+    } else {
+      args.unshift(calleeStr);
+    }
+  }
+  return args;
+}
+/**
+ * Parses and returns info about the call stack at the given index.
+ */
```

Comment:
To improve readability and maintainability, consider extracting the logic for retrieving the user email into a separate function. This will make the code cleaner and easier to test.


    *This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*